python/enthic/app.py

```diff
@@ -178,21 +178,23 @@ def pre_cast_integer(probe):
     return str(probe) if probe.__class__ is int else probe if probe.isnumeric() is True else None
 
 
-def result_array(probe, limit, offset=0):
+def result_array(probe, limit, apecode, offset=0):
     """
     List the result of the search in the database.
 
        :param probe: A string to match.
        :param limit: Integer, the limit of result to return.
+       :param ape: An integer to match
        :param offset: Integer, offset of the select SQL request.
     """
     with application.app_context():
         companies = fetchall("""SELECT siren, denomination, ape, postal_code, town
-                        FROM identity WHERE siren = %s
-                        OR denomination LIKE %s
-                        OR MATCH(denomination) AGAINST (%s IN NATURAL LANGUAGE MODE)
-                        LIMIT %s OFFSET %s;""", (pre_cast_integer(probe), "{0}%".format(probe),
-                                                 "{0}%".format(probe), limit, offset))
+                        FROM identity WHERE (siren = %s and ape regexp %s)
```

Comment:
Ape code are not directly store in the database. They are strings and we only want to store in base integer if possible, if there is a defined number of possibility it replace enum that are present in MySQL.
Hence you have to make a conversion between those integer in that are in the ontology.py and those strings.
String form request -> Integer to search in the base.

Reply:
Okay